### PR TITLE
Adding codecov github action

### PR DESCRIPTION
**Pull request by jknight-liatrio**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      - uses: actions/checkout@v2
+
+      - name: Download Dependencies
+        run: go mod tidy
+
+      - name: CodeCov
+        run: |
+          bash -c '/bin/bash <(curl -s https://codecov.io/bash)'
+
+#     - name: Build
```
Comment (mrparkers) [inline .github/workflows/test.yml]: this should be uncommented and moved before the codecov action.  can't get test coverage if we don't run tests

**Comment by mrparkers** [comment]
Let's also add the codecov build badge to the README

File: .github/workflows/test.yml (hunk 1 above)
Comment (aalsabag) [reply .github/workflows/test.yml]: can't run tests, if we don't have tests 😆 
i'll try to write a couple of tests here at the end of the day if i have time

Comment (jknight-liatrio) [reply .github/workflows/test.yml]: I think I'll draft or close this PR until we're ready to actually run some tests.

**Comment by alexashley** [comment]
Closing for now until we have some tests for this collector

**Pull request closed without merging**